cart/domain/cart/cart.go

```diff
@@ -286,6 +286,19 @@ func (c Cart) ProductCount() int {
 	return count
 }
 
+// ProductCountUnique returns the amount of unique products across all deliveries
+func (c Cart) ProductCountUnique() int {
+	marketplaceCodes := make(map[string]string)
```

Comment:
you could use `make(map[string]struct{})`, since an empty struct doesn't take up memory. 

```diff
- marketplaceCodes := make(map[string]string)
+ marketplaceCodes := make(map[string]struct{})
```

```diff
- marketplaceCodes[item.MarketplaceCode] = item.MarketplaceCode
+ marketplaceCodes[item.MarketplaceCode] = struct{}{}
```

Reply:
RThx for the hint. Learned something new. Added the suggested improvment.